src/ray/common/buffer.h

```diff
@@ -67,9 +67,10 @@ class LocalMemoryBuffer : public Buffer {
 
   /// Construct a LocalMemoryBuffer of all zeros of the given size.
   LocalMemoryBuffer(size_t size) : has_data_copy_(true) {
-    buffer_.resize(size, 0);
+    // ensure data_ won't be null_ptr when size == 0
+    buffer_.resize(std::max<size_t>(1, size), 0);
```

Comment:
This seems like a bit of a hacky workaround to me. Is the issue that the `std::vector` might have a `nullptr` underlying if size is 0? Maybe we should just enforce that `data_` is `nullptr` if `size_ == 0` instead and handle that elsewhere?

Reply:
How about handle in `write_serialized_object`:
```
    if isinstance(serialized_object, RawSerializedObject):
        if buf.get().Size() > 0:
            buffer = Buffer.make(buf)
            # `pyarrow.py_buffer` will crash when underlying buffer is a null_ptr
            stream = pyarrow.FixedSizeBufferWriter(pyarrow.py_buffer(buffer))
            stream.set_memcopy_threads(MEMCOPY_THREADS)
            stream.write(pyarrow.py_buffer(serialized_object.value))
```